### PR TITLE
Django 4.2 and fix tests

### LINKED ISSUE #354: TypeError: can't compare offset-naive and offset-aware datetimes While cheching account is expred
File "/home/ge-39/projects/python/django/sesio/env/lib/python3.6/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/home/ge-39/projects/python/django/sesio/env/lib/python3.6/site-packages/django/utils/deprecation.py", line 113, in __call__
    response = self.process_request(request)
  File "/home/ge-39/projects/python/django/sesio/env/lib/python3.6/site-packages/account/middleware.py", line 75, in process_request
    if check_password_expired(request.user):
  File "/home/ge-39/projects/python/django/sesio/env/lib/python3.6/site-packages/account/utils.py", line 137, in check_password_expired
    if expiration < now:
TypeError: can't compare offset-naive and offset-aware datetimes

### PR DESCRIPTION
**Pull request by blag**
Closes #354, but also does more.

Supercedes #369.

This PR is prep for a release. I'd like to get CI passing before I submit a PR to bump the version as part of a release.

Changes proposed in this PR:

- Move `accounts/tests` to `./tests` (so we don't distribute tests with package)
- Use timezone-aware `now()` in middleware
- Fix tests
- Use ruff
- Drop Django 2.2, and Python 3.6 and 3.7
- Update GHA config

**Tips for an ideal PR**
* You have read our Code of Conduct: https://github.com/pinax/.github/blob/master/CODE_OF_CONDUCT.md
* You have read our Contributing info: https://github.com/pinax/.github/blob/master/CONTRIBUTING.md
* The PR is atomic
* Unit tests have been updated/added, if needed
* App version number has been updated in `setup.py` (Pinax uses [SemVer](https://semver.org/))
* `Change Log` has been updated
* You have added your name to the `AUTHORS` file

### CODE REVIEW

**Comment by blag** [comment]
Okay, I think I figured out DeepSource. This is ready for review. I'll merge it once I get a signoff. 😄

**Review by uhurusurfa — APPROVED** [review]
🚀 looks good

**Pull request merged**
